### ISSUE TITLE
@property on subclass raises error

### ISSUE THREAD

**Issue by gibiansky** [python/mypy#8912]
Defining a property in a superclass and then using a `@property` decorator in a subclass does not detect that the signatures match.

I _think_ this is a bug, but I'm not sure if there are some edge cases I haven't thought through that make this error valid.

Here's code:
```python
class Superclass:                    
    prop: int                   
                                     
class Subclass(Superclass):          
    @property                        
    def prop(self) -> int: ...
```

The mypy error:
```
test.py:10: error: Signature of "prop" incompatible with supertype "Superclass"
```

Mypy `--version` is `mypy 0.770`.  

I tried to look through the list of issues with `@property` but couldn't determine if this is a duplicate; the others (that seem largely resolved) have to do with `@property` whereas this has to do with `@property` _when in a subclass with a superclass that has a type annotation_.

This _might_ be a duplicate of https://github.com/python/mypy/issues/5936; if so, please close this. I wasn't sure because that seems to have to do with setters, which this does not.

**Comment by JelleZijlstra** [comment]
This is because `prop` is assignable on the superclass but not the subclass.

Example:
```
def f(x: Superclass):
    x.prop = 42

f(Subclass())  # fails at runtime, but mypy is OK with it
```

**Comment by gibiansky** [comment]
Ah, thank you. Could there be a clearer error message for this? 

Also, should this work then:

```python
class Superclass:                    
    prop: int                        
                                     
                                     
class Subclass(Superclass):          
    @property                        
    def prop(self) -> int:           
        ...                          
                                     
    @prop.setter                     
    def prop(self, val: int) -> None:
        ...                          
```

Because I get the same errors there, but that addresses this concern.

**Comment by JelleZijlstra** [comment]
Hm, yes. That should be allowed unless I'm missing something.

Technically it's unsafe because the base class would allow `del obj.prop` and the subclass wouldn't, but I don't think mypy is that pedantic.

**Comment by rik** [comment]
I think this is a duplicate of https://github.com/python/mypy/issues/4125

**Comment by JukkaL** [comment]
Yeah, looks like a duplicate of #4125.

**Comment by gibiansky** [comment]
Thanks for figuring this out!
